### ISSUE TITLE
Move to node-coap-client

### ISSUE THREAD

**Issue by hardillb** [nidayand/node-red-contrib-tradfri#15]
Hi,

I've been looking at this since the lights shipped and I've been planning a similar node (since I did the WeMo node), I've been waiting on a working nodejs native coap client to show up that should make it much more portable.

The [node-coap-client](https://www.npmjs.com/package/node-coap-client) looks like it's finally got to the point where it's stable and usable. Have you had any thoughts about moving to it rather than libcoap?

**Comment by nidayand** [comment]
I haven't really done anything for quite some time due to other projects.. but if I get the time and energy I will see if that is a valid option  or if others will chip with a migration. I would like to get it properly verified that it is a good idea first though as the coap binary has been around for quite some time and is well proven.

**Comment by nidayand** [comment]
I'm currently working on moving from the argon library to node-tradfri-client library and consequently it will only support nodejs v6 (which is now the officially supported version of nodered)

**Comment by ruudboon** [comment]
Awesome @nidayand looks like node-tradfri-client also have the possibility to support scenes and subscribe to the current status. Any estimates on you timeline? 
Thank you for this project!

**Comment by nidayand** [comment]
Hi @ruudboon . Unfortunately it has a low priority right now and I cannot give an estimate. I've started the conversion but it is not ready for a commit..

**Comment by pedesen** [comment]
@nidayand node-tradfri-client looks indeed promising, especially the possibility to subscribe, which would make it possible to react on button presses on a tradfri remote, as far as I understood. Let me know if I can contribute.

**Comment by RaymondMouthaan** [comment]
Any news regarding this enhancement ?
